### ISSUE TITLE
Structcheck linter stopped working with generic types in 1.53.0

### ISSUE THREAD

**Issue by pavelpatrin** [golangci/golangci-lint#3881]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [ ] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Structcheck linter started reporting false-positive results for generic structs in 1.53.0. In 1.52.0 everything worked fine.

```
myservice/models/unitofwork/unitofwork.go:83:2: `appliesTimer` is unused (structcheck)
	appliesTimer   prometheus.Observer
	^
```

I think abandoned linter should be deleted. 

### Version of golangci-lint

<details>
1.53.1
</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
run:
  tests: false
  go: '1.18'

linters:
  disable-all: true
  enable:
    - structcheck
```

</details>

### Go environment

<details>

```console
$ go version && go env
# go version go1.20 darwin/arm64
```
</details>

### Verbose output of running

<details>

```

```

</details>

### Code example or link to a public repository

<details>

```

```

</details>

**Comment by ldez** [comment]
Hello,

can you provide the output of `golangci-lint version`?

**Comment by ldez** [comment]
structcheck is deprecated, so yes it doesn't work as expected.
